nixos/modules/services/cluster/druid/default.nix

```
@@ -0,0 +1,225 @@
+        "druid.extensions.loadList" = ''[ "mysql-metadata-storage" ]'';
+      };
+    };
+    commonTmpDirs = mkOption {
+      default = [ "/var/log/druid/requests" ];
+      type = types.listOf types.str;
+      description = "Common List of directories used by druid processes";
+    };
+    log4j = mkOption {
+      type = types.path;
+      description = "Log4j Configuration for the druid process";
+    };
+    extraClassPaths = mkOption {
+      default = [ ];
+      type = types.listOf types.str;
+      description = "Extra classpath to include in the jvm";
+    };
+    extraConfDirs = mkOption {
+      default = [ ];
+      type = types.listOf types.path;
+      description = "Extra Conf Dirs to include in the jvm";
+    };
+    overlord = druidServiceOption "Druid Overlord";
+    coordinator = druidServiceOption "Druid Coordinator";
+    broker = druidServiceOption "Druid Broker";
+    historical = (druidServiceOption "Druid Historical") // {
+      segmentLocations = mkOption {
+        default = null;
+        description = "Locations where the historical will store its data.";
+        type = with types; nullOr (listOf (submodule {
+          options = {
+            path = mkOption {
+              type = types.path;
+              description = "the path to store the segments";
+            };
+            maxSize = mkOption {
+              type = types.str;
+              description = "Max size the druid historical can occupy";
+            };
+            freeSpacePercent = mkOption {
+              type = types.float;
+              description = "Druid Historical will fail to write if it exceeds this value";
+            };
+          };
+        }));
+      };
+    };
+    middleManager = druidServiceOption "Druid middleManager";
+    router = druidServiceOption "Druid Router";
+  };
+  config = mkMerge [
+    (druidServiceConfig rec {
+      name = "overlord";
+      allowedTCPPorts = [ (attrByPath [ "druid.plaintextPort" ] 8090 cfg."${name}".config) ];
+    })
+    (druidServiceConfig rec {
+      name = "coordinator";
+      allowedTCPPorts = [ (attrByPath [ "druid.plaintextPort" ] 8081 cfg."${name}".config) ];
+    })
+    (druidServiceConfig rec {
+      name = "broker";
+      tmpDirs = [ (attrByPath [ "druid.lookup.snapshotWorkingDir" ] "" cfg."${name}".config) ];
+      allowedTCPPorts = [ (attrByPath [ "druid.plaintextPort" ] 8082 cfg."${name}".config) ];
+    })
+    (druidServiceConfig rec {
+      name = "historical";
+      tmpDirs = [ (attrByPath [ "druid.lookup.snapshotWorkingDir" ] "" cfg."${name}".config) ]
+        ++ (map (x: x.path) cfg."${name}".segmentLocations);
+      allowedTCPPorts = [ (attrByPath [ "druid.plaintextPort" ] 8083 cfg."${name}".config) ];
+      extraConfig.services.druid.historical.internalConfig."druid.segmentCache.locations" = ''
+        [${(concatMapStringsSep "," (
+          x: "{\"path\":\"" + x.path + "\",\"maxSize\":" + x.maxSize + "}"
```

```suggestion
          x: ''{"path":"${x.path}","maxSize":"${x.maxSize}"}''
```
[optional] get rid of all the escapes